checks.d/vsphere.py

```diff
@@ -748,7 +748,13 @@ def _collect_metrics_atomic(self, instance, mor):
                     continue
                 instance_name = result.id.instance or "none"
                 value = self._transform_value(instance, result.id.counterId, result.value[0])
-                self.gauge(
+
+                # Most common metric types are absolute, delta, and rate
+                if self.metrics_metadata[i_key][result.id.counterId]['s_type'] == 'rate':
```

Comment:
I think this would throw a `KeyError` exception, from what I saw in the check the metadata object would only have `name`, `unit` and `instance_tag` keys.